src/main/java/org/reactome/release/downloaddirectory/MapOldStableIds.java

```diff
@@ -79,50 +83,107 @@ public static void execute(String username, String password, String host, String
+		Set<String> currentStableIdentifiersSet = new HashSet<>();
+		for (GKInstance stableIdentifierInst : stableIdentifierInstances) {
+			currentStableIdentifiersSet.add(stableIdentifierInst.getAttributeValue(ReactomeJavaConstants.identifier).toString());
+		}
+		return currentStableIdentifiersSet;
+	}
+
+	/**
+	 * Uses the resultSet from the stable_identifiers database query (which retrieved *all* StableIdentifiers and
+	 * their associated instance ids that have ever existed in Reactome) to build a map of instance IDs to StableIdentifiers.
+	 * @param stableIdResults ResultSet - Data result of query to stable_identifiers database for stable identifiers and associated instance IDs.
+	 * @return Map<String, List<String>> - Mapping of db IDs to Stable Identifiers.
+	 * @throws SQLException - Thrown if there are issues accessing the ResultSet object.
+	 */
+	private static Map<String, List<String>> getDbIdToStableIds(ResultSet stableIdResults) throws SQLException {
 		Map<String, List<String>> dbIdToStableIds = new HashMap<>();
 
 		// Iterate through returned results of DB IDs and stable IDs
-		while (resultSet.next()) {
-			String stableId = resultSet.getString(1);
-			String dbId = resultSet.getString(2);
+		while (stableIdResults.next()) {
+			String stableId = stableIdResults.getString(1);
+			String dbId = stableIdResults.getString(2);
 
-			if (dbIdToStableIds.get(dbId) != null) {
-				dbIdToStableIds.get(dbId).add(stableId);
-			} else {
-				List<String> stableIds = new ArrayList<>(Collections.singletonList(stableId));
-				dbIdToStableIds.put(dbId, stableIds);
-			}
+			dbIdToStableIds.computeIfAbsent(dbId, k -> new ArrayList<>()).add(stableId);
 		}
-
 		return dbIdToStableIds;
 	}
+
+	/**
+	 * With the old stable identifier mappings completed, write the results to the 'reactome_stable_ids.txt' file.
+	 * @param releaseNumber String - Current release number, used for storing files in release-specific location.
+	 * @param stableIdsToOldIdsMappings List<List<Object>> - List of current stable identifier mappings to older mappings.
+	 * The interior List<Object> is of the form [String, List<String>].
+	 * @param currentStableIdentifiers Set<String>, all StableIdentifiers in current release database.
+	 * @throws IOException - Thrown if there are issues with creating or moving mapping file.
+	 */
+	private static void writeMappingsToFile(String releaseNumber, List<List<Object>> stableIdsToOldIdsMappings, Set<String> currentStableIdentifiers) throws IOException {
+		String header = "# Reactome stable IDs for release " + releaseNumber + "\n" + "Stable_ID\told_identifier(s)\n";
+		Files.write(Paths.get(filename), header.getBytes(), StandardOpenOption.APPEND);
+		for (List<Object> stableIdsArray : stableIdsToOldIdsMappings)
+		{
+			String primaryId = (String) stableIdsArray.get(0);
+			@SuppressWarnings("unchecked")
+			List<String> secondaryIds = (ArrayList<String>) stableIdsArray.get(1);
```

Comment:
Just to clarify: stableIdsArray[0] is a scalar (a String), but stableIdsArray[1] is another nested list?

I'm wondering if this is one of those times where it makes sense to create a simple private class within this class that stores structure. Something like
```Java
class StabldIdMapping {
    String primaryId;
    List<String> secondaryIds;
//...
}
```
Then you could have a `List<StabldIdMapping>`. Not 100% sure this will work, but something to consider.

Java's not very good at these sorts of structures, but hopefully JEP 359 (https://openjdk.java.net/jeps/359) will help resolve that.

Reply:
I think a custom object does make sense here.  I have a bit of trouble understanding the nested list structure since different object types are held in the list.

Reply:
I like the idea of creating a custom object, I am going to kick the can down the road a little bit and create an issue for it rather than implement it here, but I will implement it since it makes more sense than just having an Object